src/node_sqlite.cc

```diff
@@ -128,6 +149,169 @@ inline void THROW_ERR_SQLITE_ERROR(Isolate* isolate, int errcode) {
   isolate->ThrowException(error);
 }
 
+class BackupJob : public ThreadPoolWork {
+ public:
+  explicit BackupJob(Environment* env,
+                     DatabaseSync* source,
+                     Local<Promise::Resolver> resolver,
+                     std::string source_db,
+                     std::string destination_name,
+                     std::string dest_db,
+                     int pages,
+                     Local<Function> progressFunc)
+      : ThreadPoolWork(env, "node_sqlite3.BackupJob"),
+        env_(env),
+        source_(source),
+        source_db_(source_db),
+        destination_name_(destination_name),
+        dest_db_(dest_db),
+        pages_(pages) {
+    resolver_.Reset(env->isolate(), resolver);
+    progressFunc_.Reset(env->isolate(), progressFunc);
+  }
+
+  void ScheduleBackup() {
+    Isolate* isolate = env()->isolate();
+    HandleScope handle_scope(isolate);
+
+    backup_status_ = sqlite3_open(destination_name_.c_str(), &pDest_);
+
+    Local<Promise::Resolver> resolver =
+        Local<Promise::Resolver>::New(env()->isolate(), resolver_);
+
+    Local<Object> e = Local<Object>();
+
+    if (backup_status_ != SQLITE_OK) {
+      e = CreateSQLiteError(isolate, pDest_).ToLocalChecked();
```

Comment:
```suggestion
    if (!CreateSQLiteError(isolate, pDest_).ToLocal(&e)) {
      Cleanup();
      return;
    }
```

Reply:
Likewise below. Just trying to minimize use of `ToLocalChecked` since that will crash the process.